backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/handlingunits/rest_api/HandlingUnitsService.java

```diff
@@ -519,4 +570,91 @@ private HuId resolveHUIdentifier(@NonNull final JsonSetClearanceStatusRequest.Js
 				.appendParametersToMessage()
 				.setParameter("huIdentifier", jsonHuIdentifier);
 	}
+
+	@NonNull
+	private Set<HuId> resolveHuIds(@NonNull final JsonGetByQRCodeRequest request)
+	{
+		final HUQRCodeAssignment huqrCodeAssignment = huQRCodeService
+				.getHUAssignmentByQRCode(HUQRCode.fromGlobalQRCodeJsonString(request.getQrCode()))
+				.orElse(null);
+
+		if (huqrCodeAssignment == null)
+		{
+			return ImmutableSet.of();
+		}
+
+		if (request.getUpperLevelLocatingQrCode() == null)
+		{
+			return huqrCodeAssignment.getHuIds();
+		}
+
+		return resolveHuIdsBy(GlobalQRCode.ofString(request.getUpperLevelLocatingQrCode()), huqrCodeAssignment);
+	}
+
+	@NonNull
+	private Set<HuId> resolveHuIdsBy(@NonNull final GlobalQRCode locatingQrCode, @NonNull final HUQRCodeAssignment huqrCodeAssignment)
+	{
+		if (HUQRCode.isTypeMatching(locatingQrCode))
+		{
+			return resolveHuIdsBy(HUQRCode.fromGlobalQRCode(locatingQrCode), huqrCodeAssignment);
+		}
+		else if (LocatorQRCode.isTypeMatching(locatingQrCode))
+		{
+			return resolveHuIdsBy(LocatorQRCode.ofGlobalQRCode(locatingQrCode), huqrCodeAssignment);
+		}
+		else
+		{
+			throw new AdempiereException("Unknown type for qrCode")
+					.appendParametersToMessage()
+					.setParameter("qrCode", locatingQrCode);
+		}
+	}
+
+	@NonNull
+	private Set<HuId> resolveHuIdsBy(@NonNull final LocatorQRCode locatorQRCode, @NonNull final HUQRCodeAssignment targetQrCodeAssignment)
+	{
+		return handlingUnitsDAO.createHUQueryBuilder()
+				.addOnlyHUIds(targetQrCodeAssignment.getHuIds())
+				.addOnlyInLocatorId(locatorQRCode.getLocatorId())
+				.setOnlyTopLevelHUs(false)
+				.listIds();
+	}
+
+	@NonNull
+	private Set<HuId> resolveHuIdsBy(@NonNull final HUQRCode parentHUQrCode, @NonNull final HUQRCodeAssignment targetQrCodeAssignment)
```

Comment:
rename it to resolveHuIdsByHUQRCode

Reply:
wondering if this method shall go to some other better service ....

Reply:
thought about it, but not sure where it would fit better than HUService :-?...